pkg/visitors/assemble.go

```diff
@@ -71,6 +72,86 @@ func (v *Assemble) Visit(f uefi.Firmware) error {
 		}
 
 		fileOffset := f.DataOffset
+
+		var pad *uefi.File
+		var shrink uint64
+		for _, file := range f.Files {
+			fileBuf := file.Buf()
+			fileLen := uint64(len(fileBuf))
+			if fileLen == 0 {
+				log.Fatal("Unexpected empty file buffer (at least a header was expected): %v", file.Header.GUID)
+			}
+			if file.Header.GUID == *uefi.FFGUID && fileLen > 24 {
+				// found non-empty PAD to check for free space.
```

Comment:
how do you know this pad file is non empty? it could be empty.

Reply:
I'll have to check. what defines an non-empty pad file UEFITool said it was and that is the file that I'm trying to modify.  I'm not trying to modify all pad files but the non-empty pad file that is just before the volume top file.  I need to look again at the details of this since in my use model, only the non-empty pad file is adjusted.  For my example, pad-file are ignored and only the non-empty one is shrunk.